pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -107,17 +107,24 @@ func (r *Reconciler) Reconcile(ctx context.Context) error {
 	return nil
 }
 
-func (r *Reconciler) unmapDynakube(ctx context.Context) {
-	if meta.FindStatusCondition(*r.dk.Conditions(), codeModulesInjectionConditionType) != nil &&
-		meta.FindStatusCondition(*r.dk.Conditions(), metaDataEnrichmentConditionType) != nil {
+func (r *Reconciler) cleanup(ctx context.Context) {
+	if meta.FindStatusCondition(*r.dk.Conditions(), codeModulesInjectionConditionType) == nil &&
```

Comment:
The inverted nil check makes total sense here. Out of curiosity: why was there a `not equal` check before? Am I missing something?

Reply:
probably a bug, I see no logic in the previous one 😅